data/tutorials/getting-started/gs_01_a_tour_of_ocaml.md

```diff
@@ -0,0 +1,874 @@
+```
+
+Note that exceptions do not appear in function types.
+
+Exceptions are caught using the `try … with …` construction
+
+```ocaml
+# try id_42 0 with Failure _ -> 0;;
+- : int = 0
+```
+
+The standard library provides several predefined exceptions. It is possible to define exceptions.
+
+### Using the `result` type
+
+Another way to deal with errors in OCaml is by returning value of type `result`,
+which can represent either the correct result or an error. Here is how it is defined:
+
+```ocaml
+# #show result;;
+type ('a, 'b) result = Ok of 'a | Error of 'b
+```
+
+So one may write:
+
+```ocaml
+# let id_42_res n = if n <> 42 then Error "Sorry" else Ok n;;
+val id_42_res : int -> (int, string) result = <fun>
+
+# id_42_res 42;;
+- : (int, string) result = Ok 42
+
+# id_42_res 0;;
+- : (int, string) result = Error "Sorry"
+
+# match id_42_res 0 with
+  | Ok n -> n
+  | Error _ -> 0;;
+- : int = 0
+```
+
+## Working with Mutable State
+
+OCaml supports imperative programming. Usually, the `let … = …` syntax does not define variables, it defines constants. However, mutable variables exists in OCaml, they are called references. Here's how we create a reference to an integer:
+
+```ocaml
+# let r = ref 0;;
+val r : int ref = {contents = 0}
+```
+
+It is syntactically impossible to create an unintialised or null reference. The `r`
+reference is created with the integer zero. Accessing a reference's content
+is done using the `!` de-reference operator.
+
+```ocaml
+# !r;;
+- : int = 0
+```
+
+Let's put something else behind `r`. Here `:=` is the assignment operator; it is
+pronounced “receives”.
+
+```ocaml
+# r := 42;;
+- : unit = ()
+```
+
+This returns `()` because changing the content of a reference is a side-effect.
+
+```ocaml
+# !r;;
+- : int = 42
+```
+
+Execute an expression after another with the `;` operator. Writing `a; b` means: execute `a`; once done, execute `b`, only return the value of `b`.
+
+```ocaml
+# let text = ref "hello ";;
+val text : string ref = {contents = "hello "}
+
+# print_string !text; text := "world!"; print_endline !text;;
+hello world!
+- : unit = ()
+```
+
+Here are the side effects which happens in the second line:
+1. Display the contents of the reference `text` on standard output
+1. Update the contents of the reference  `text`
+1. Display the contents of the reference `text` on standard output
+
+This behaviour is the same as in an imperative language. However, although `;` is not defined as a function, it behaves as if it was a function of type `unit -> unit -> unit`.
+
+OCaml also has mutable arrays. Here is how to create one:
+
+```ocaml
+# let a = [| 11; 12; 13; 14; 15; 16; 17 |];;
+val a : int array = [|11; 12; 13; 14; 15; 16; 17|]
+```
+
+Here is how indexed access is done:
+
+```ocaml
+# a.(4);;
+- : int = 15
+```
+
+Indexed assignment is done using the `<-` operator (not the reference assignment `:=`).
+```ocaml
+# a.(4) <- 0;;
+- : unit = ()
+
+# a.(4);;
+- : int = 0
+```
+
```

Comment:
```suggestion

```

Thibaut suggested we move this to another tutorial and put a link here, something like:

When you're ready, explore other topics like [dealing with errors](link), [working in a mutable state](link), and using [operators as functions](link) 

(as an example, please feel free to revise!)